File: dcmtk/src/main/java/com/example/dcmtk/model/PixelData.java
package com.example.dcmtk.model;

public class PixelData {
    private int rows;
    private int columns;
    private byte[] data;
    private int largestImagePixelValue;
    private int win_center;
    private int win_width;
    private final int exposure_leve;
    private final double standardDeviation;

    public PixelData(int rows, int columns, byte[] data, int largestImagePixelValue
            , int win_center, int win_width, int exposure_leve, double standardDeviation) {
        this.rows = rows;
        this.columns = columns;
        this.data = data;
        this.largestImagePixelValue = largestImagePixelValue;
        this.win_center = win_center;
        this.win_width = win_width;
        this.exposure_leve = exposure_leve;
        this.standardDeviation = standardDeviation;
    }

    public double getStandardDeviation() {
        return standardDeviation;
    }

    public int getExposure_leve() { return exposure_leve;}

    public int getRows() {
        return rows;
    }

    public void setRows(int rows) {
        this.rows = rows;
    }

    public int getColumns() {
        return columns;
    }

    public void setColumns(int columns) {
        this.columns = columns;
    }

    public byte[] getData() {
        return data;
    }

    public void setData(byte[] data) {
        this.data = data;
    }
    public int getLargestImagePixelValue() {
        return largestImagePixelValue;
    }
    public void setLargestImagePixelValue(int largestImagePixelValue) {
        this.largestImagePixelValue = largestImagePixelValue;
    }

    public int getWin_center() {
        return win_center;
    }

    public void setWin_center(int win_center) {
        this.win_center = win_center;
    }

    public int getWin_width() {
        return win_width;
    }

    public void setWin_width(int win_width) {
        this.win_width = win_width;
    }
}

File: dcmtk/src/main/java/com/example/dcmtk/model/PixelData.kt
package com.example.dcmtk.model

/**
 * 像素数据处理结果。
 * 字段顺序即 JNI 构造器签名顺序：(I I [B D D I I)，native 构造 PixelData 时须严格匹配。
 */
data class PixelData(
    val rows: Int,
    val columns: Int,
    val data: ByteArray,
    val win_width: Double,
    val win_center: Double,
    val exposure_leve: Int,
    val largestImagePixelValue: Int
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is PixelData) return false
        return rows == other.rows && columns == other.columns &&
                win_width == other.win_width && win_center == other.win_center &&
                exposure_leve == other.exposure_leve &&
                largestImagePixelValue == other.largestImagePixelValue &&
                data.contentEquals(other.data)
    }

    override fun hashCode(): Int {
        var r = rows
        r = 31 * r + columns
        r = 31 * r + data.contentHashCode()
        r = 31 * r + win_width.hashCode()
        r = 31 * r + win_center.hashCode()
        r = 31 * r + exposure_leve
        r = 31 * r + largestImagePixelValue
        return r
    }
}
